src/core/internal/elf/io.d

```diff
@@ -349,23 +350,128 @@ private struct MMapRegion
     public const(ubyte)[] data;
 }
 
+@nogc nothrow:
+
+version (OpenBSD)
+private extern(C) const(char)* getprogname();
+
+/// Returns the path to the process' executable as newly allocated C string
+/// (free() when done), or null on error.
+version (LinuxOrBSD)
+char* thisExePath()
+{
+    version (linux)
+    {
+        return readLink("/proc/self/exe");
+    }
+    else version (Solaris)
+    {
+        import core.stdc.stdio : snprintf;
+        import core.sys.posix.unistd : getpid;
+
+        // only Solaris 10 and later
+        char[32] buffer = void;
+        const numWritten = snprintf(buffer.ptr, sizeof(buffer), "/proc/%d/path/a.out", getpid());
+        assert(numWritten > 0 && numWritten < sizeof(buffer));
+        assert(buffer[numWritten] == 0);
+
+        return readLink(buffer.ptr);
+    }
+    else version (OpenBSD)
+    {
+        import core.stdc.string : strdup;
+        return strdup(getprogname()); // FIXME
```

Comment:
Phobos' OpenBSD version is quite messy and involves other Phobos helpers: https://github.com/dlang/phobos/blob/40aebd4c34802377a9dcb8a1f3c96ce311b34e0e/std/file.d#L3557-L3594

This simple version is what's currently used and so doesn't fix the issue on OpenBSD.

Reply:
(Pinging @ibara.)

Reply:
There is no way to get the pathname of a running executable on OpenBSD. This is the best that can be done.